code/game/objects/items/reagent_containers/food/cans.dm

```diff
@@ -1,29 +1,94 @@
 /obj/item/reagent_container/food/drinks/cans
-	var/canopened = FALSE
+	var/open = FALSE
+	//If it needs can opener to be opened
+	var/needs_can_opener = FALSE
 	var/crushed = FALSE
+	//Can be crushed
+	var/crushable = TRUE
+	//Can open sound
+	var/open_sound = 'sound/effects/canopen.ogg'
+	//Can open message
+	var/open_message = "You open the drink with an audible pop!"
+	//Eating sound
+	var/consume_sound = 'sound/items/drink.ogg'
+	//What this object is, used during interactions
+	var/object_fluff = "drink"
+	//If can transfer reagents to food
+	var/food_interactable = FALSE
+	//If can has a dedicated crushed icon
+	var/crushed_icon = null
+	//If can has a dedicated open icon
+	var/has_open_icon = FALSE
+	//Should item be deleted on being empty
+	var/delete_on_empty = FALSE
 	gulp_size = 10
 	icon = 'icons/obj/items/food/drinkcans.dmi'
 	item_icons = list(
 		WEAR_L_HAND = 'icons/mob/humans/onmob/inhands/items/food_lefthand.dmi',
 		WEAR_R_HAND = 'icons/mob/humans/onmob/inhands/items/food_righthand.dmi'
 	)
 
+/obj/item/reagent_container/food/drinks/cans/attackby(obj/item/opening_tool as obj, mob/user as mob)
+	var/opening_time
+	var/opening_sound
+	var/hiss = pick("Nice hiss!", "No hiss.", "A small hiss.") //i couldn't not include stevemre reference
+	if(user.action_busy || open || !needs_can_opener || !HAS_TRAIT(opening_tool, TRAIT_TOOL_CAN_OPENER))
+		return
+
+	if(istype(opening_tool, /obj/item/tool/kitchen/can_opener))
+		opening_time = 4 SECONDS
+		opening_sound = 'sound/items/can_open2.ogg'
+		to_chat(user, SPAN_NOTICE("You begin to open the can with a can opener. [hiss]"))
+	if(istype(opening_tool, /obj/item/attachable/bayonet || /obj/item/tool/kitchen/knife  || /obj/item/weapon/sword/machete))
+		opening_time = 12 SECONDS
+		opening_sound = 'sound/items/can_open1.ogg'
+		to_chat(user, SPAN_NOTICE("You begin to crudely jam the can with a blade. [hiss]"))
+
+	playsound(src.loc, opening_sound, 15, FALSE, 5)
+
+	if(do_after(user, opening_time, INTERRUPT_ALL, BUSY_ICON_GENERIC))
+		if(prob(25) && istype(opening_tool, /obj/item/attachable/bayonet || /obj/item/tool/kitchen/knife  || /obj/item/weapon/sword/machete))
```

Comment:
you can't do this (/obj/item/attachable/bayonet || /obj/item/tool/kitchen/knife || /obj/item/weapon/sword/machete)